### ISSUE TITLE
Error Running Nestjs App with Prisma

### ISSUE THREAD

**Issue by walidboulanouar** [prisma/prisma#16805]
### Bug description

Hello Prisma Community, I really had a problem when i want to run my nestjs+prisma app on cpanel, i am struggling with this many days without having any resource or any help in stack overflow or any platform, I am really build my app fastly using nest and prisma compare to other technologies.
But the problem is when i want to deploy my app on cpanel, i need to run npx prisma db push to generate files on node modules, to make my app running properly. but it shows me that error
![image](https://user-images.githubusercontent.com/72222865/207641132-3607ad23-cfb4-4ae2-990f-76a5de9ce39b.png)

### Expected behavior

npx prisma db push should generate prisma client files

### Prisma information

![image](https://user-images.githubusercontent.com/72222865/207641175-32c3f09d-01bb-44b2-b4fa-370be4ac57ea.png)

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

4.3.1

**Comment by janpio** [comment]
Your application works successfully locally?
Where are you hosting your cpanel?
Have you tried hosting somewhere else?

**Comment by walidboulanouar** [comment]
yeah it is working perfectely untill i deploy it in cpanel, i am hosting this app in cpanel like hostinger, i hosted my app in digital ocean but so expensive , so i want to host it in cpanel. if some one succeffully do that in cpanel may help me in the step of generation the files, npx prisma db push or generate

**Comment by walidboulanouar** [comment]
i will try to add ram right?

**Comment by walidboulanouar** [comment]
this is the  complete log

**Comment by janpio** [comment]
The error message is telling you there is not enough memory to run the command. Something in cpanel is limiting the memory the process has, and it is not enough to run Prisma. This is almost never a problem (actually, you are the first one that reports this specific one!) so that environment needs to be really memory constrained.

There is unfortunately nothing we can do. Use any other hosting and see if this works.

**Comment by walidboulanouar** [comment]
thank you for your answer, i hope that the problem is on cpanel environment, i will see with other alternative

**Comment by walidboulanouar** [comment]
i test it on other cpanel, other platform, this is the same problem

**Comment by janpio** [comment]
If cpanel limits the memory of this script step too much, that is expected. Maybe try to find out how much resources are available in that situation.

Can you describe exactly your setup? Then we can try to reproduce this ourselves.

**Comment by walidboulanouar** [comment]
i contact the support also they offer me 12 go ram, but nothing happen
![image](https://user-images.githubusercontent.com/72222865/207881060-338761fb-b6fc-432b-85a1-6783a2d4ca45.png)
my cpanel is running on top of cloudlinux, why we do guide how you can run prisma in platform like cpanel, a lot of people using cpanel, but if we ignore them it will be a huge lost to prisma, because other platform dosn't store image, images losted every time the system boot but cpanel make a backup and you can control files as you want.

**Comment by janpio** [comment]
The process that is failing does definitely not have 12GB of RAM available. This should work with a few hundred Megabyte easily.

> Can you describe exactly your setup? Then we can try to reproduce this ourselves.

**Comment by walidboulanouar** [comment]
i also try to add npx --max-old-space-size=8192 but didn't work

**Comment by walidboulanouar** [comment]
i also try different package manager like : pnpx and yarn and no one of them succeed
returncode: 1
stdout:
> sedi_hisham_back_end@0.0.1 prisma
> yarn prisma generate

yarn run v1.22.19
$ yarn prisma generate  generate
$ yarn prisma generate  generate generate
$ yarn prisma generate  generate generate generate
$ yarn prisma generate  generate generate generate generate
$ yarn prisma generate  generate generate generate generate generate
$ yarn prisma generate  generate generate generate generate generate generate
$ yarn prisma generate  generate generate generate generate generate generate generate
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
stderr:
error Command failed with signal "SIGABRT".
error Command failed with exit code 1.
error Command failed with exit code 1.
error Command failed with exit code 1.
error Command failed with exit code 1.
error Command failed with exit code 1.
error Command failed with exit code 1.

**Comment by walidboulanouar** [comment]
I also installed prisma globally to use cli but also the same error as npx prisma

**Comment by janpio** [comment]
I can only repeat myself now:

1. This is caused by the amount of memory your cpanel installation provides to the command that is running. It is not enough to run it.
2. If you answer my question "Can you describe exactly your setup? Then we can try to reproduce this ourselves." we can take a look ourselves. Without that response we can not and we will not be able to help you.

**Comment by walidboulanouar** [comment]
my package.json 
{
  "name": "sedi_hisham_back_end",
  "version": "0.0.1",
  "description": "",
  "author": "",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "prebuild": "rimraf dist & npx prisma db push ",
    "nest": "npm i -g @nestjs/cli",
    "prisma": "npx prisma db push",
    "build": "nest build",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
    "start": "nest start",
    "start:dev": "nest start --watch",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main",
    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json"
  },
  "dependencies": {
    "@nestjs/common": "^8.4.7",
    "@nestjs/config": "^1.2.1",
    "@nestjs/core": "^8.0.0",
    "@nestjs/jwt": "^9.0.0",
    "@nestjs/mapped-types": "*",
    "@nestjs/passport": "^9.0.0",
    "@nestjs/platform-express": "^8.4.7",
    "@nestjs/serve-static": "^2.2.2",
    "@nestjs/swagger": "^5.2.1",
    "@prisma/client": "^4.3.1",
    "@sendgrid/mail": "^7.7.0",
    "@types/bcrypt": "^5.0.0",
    "bcrypt": "^5.0.1",
    "class-transformer": "^0.5.1",
    "class-validator": "^0.13.2",
    "exceljs": "^4.3.0",
    "fork-ts-checker-webpack-plugin": "^7.2.13",
    "nodemailer": "^6.8.0",
    "passport": "^0.6.0",
    "passport-jwt": "^4.0.0",
    "passport-local": "^1.0.0",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^7.2.0",
    "tmp": "^0.2.1",
    "uuid": "^9.0.0",
    "yarn": "^1.22.19"
  },
  "devDependencies": {
    "@nestjs/cli": "^8.0.0",
    "@nestjs/schematics": "^8.0.0",
    "@nestjs/testing": "^8.0.0",
    "@types/express": "^4.17.13",
    "@types/jest": "27.5.0",
    "@types/multer": "^1.4.7",
    "@types/node": "^16.0.0",
    "@types/passport-jwt": "^3.0.7",
    "@types/passport-local": "^1.0.34",
    "@types/supertest": "^2.0.11",
    "@typescript-eslint/eslint-plugin": "^5.0.0",
    "@typescript-eslint/parser": "^5.0.0",
    "eslint": "^8.0.1",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "jest": "28.0.3",
    "prettier": "^2.3.2",
    "prisma": "^4.3.1",
    "source-map-support": "^0.5.20",
    "supertest": "^6.1.3",
    "ts-jest": "28.0.1",
    "ts-loader": "^9.2.3",
    "ts-node": "^10.0.0",
    "tsconfig-paths": "4.0.0",
    "typescript": "^4.3.5"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".*\\.spec\\.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "collectCoverageFrom": [
      "**/*.(t|j)s"
    ],
    "coverageDirectory": "../coverage",
    "testEnvironment": "node"
  }
}

**Comment by janpio** [comment]
Does this also happen when you have a minimal Prisma project (via `npx prisma init`) and then create a minimal schema file and try to `db push` that schema?

**Comment by walidboulanouar** [comment]
yeah i already try just try npx prisma --v, and the same erro, every time i try a command using prisma it shows the same error

**Comment by janpio** [comment]
How can we reproduce this ourselves? What do we have to do to see the same?

**Comment by walidboulanouar** [comment]
you can create a node js application, and add prisma to it with some schema, then add some endpoints using node js and prisma, then deploy the app in cpanel, a lot of video explain how, but if you want me to do the steps, i will record a video and put the link here

**Comment by janpio** [comment]
Please do, but only for the Cpanel steps. Where do I get a Cpanel? How exactly do I configure it to run a Node app that is using Prisma?

**Comment by ntvsp** [comment]
same issue, and i can not fix

**Comment by ntvsp** [comment]
chat GPt:
"The "Out of memory" error when using WebAssembly typically means that the JavaScript engine has run out of memory to allocate to the WebAssembly instance. There are several possible reasons why this could be happening:

The WebAssembly module is too large for the available memory. In this case, you could try reducing the size of the module by optimizing the code or reducing the amount of data it uses.

The JavaScript code that is calling the WebAssembly module is leaking memory. You could try using a memory profiler to find and fix any memory leaks in your code.

The JavaScript engine itself is running out of memory. You could try increasing the memory limit of your JavaScript engine using the --max-old-space-size flag. For example, to set the memory limit to 4 GB, you could use node --max-old-space-size=4096.

The computer running the code does not have enough available memory. In this case, you may need to upgrade the computer or run the code on a computer with more memory.

It's also worth noting that the error message can sometimes be caused by bugs in the JavaScript engine or the WebAssembly runtime, so updating to the latest versions of those components may also help."

**Comment by JohnMwanyika** [comment]
`stdout:
> daraja_api@1.0.0 generate
> prisma generate
stderr:
/home/mcrhorke/nodevenv/repositories/daraja_stk_push/16/lib/node_modules/prisma/build/index.js:12451
    var wasmInstance = new WebAssembly.Instance(wasmModule, imports);
                       ^

RangeError: WebAssembly.Instance(): Out of memory: wasm memory
    at ../../node_modules/.pnpm/@prisma+prisma-fmt-wasm@4.11.0-57.8fde8fef4033376662cad983758335009d522acb/node_modules/@prisma/prisma-fmt-wasm/src/prisma_fmt_build.js (/home/mcrhorke/nodevenv/repositories/daraja_stk_push/16/lib/node_modules/prisma/build/index.js:12451:24)
    at __require (/home/mcrhorke/nodevenv/repositories/daraja_stk_push/16/lib/node_modules/prisma/build/index.js:14:50)
    at Object.<anonymous> (/home/mcrhorke/nodevenv/repositories/daraja_stk_push/16/lib/node_modules/prisma/build/index.js:87712:38)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47`

having the same issue its been 3weeks now researching on the same no success,,this is really frustrating

**Comment by Tooluloope** [comment]
Same issue

![Screenshot 2023-05-16 at 17 24 03](https://github.com/prisma/prisma/assets/31691737/5099df61-665b-4f8a-b051-219b12074c30)

**Comment by JohnMwanyika** [comment]
Been researching on this issue with no success, it became frustrating coz my client wanted the app hosted ASAP,,I  asked them to at least allow me 2weeks,,went a long and learnt Sequelize and luckily I was able to successfully host the application.  Ever since, I stopped using Prisma

**Comment by ntvsp** [comment]
I was able to run it but I don't remember how to fix it. maybe it's because 'argon2',

**Comment by ALWISHARIFF** [comment]
Assalamu alaykum warahmatullahy wabarakatuh @walidboulanouar Did you find a way to solve it
